codex-rs/common/src/oss.rs

```diff
@@ -13,10 +14,28 @@ pub fn get_default_model_for_oss_provider(provider_id: &str) -> Option<&'static
     }
 }
 
+/// Detect whether the selected Ollama instance supports the responses API and, if not, downgrade
+/// to the chat completions wire API. This should run whenever the Ollama provider is selected,
+/// even when `--oss` is not in use, so older servers remain compatible.
+pub async fn detect_ollama_wire_api_if_needed(config: &mut Config) {
```

Comment:
I would name this `update_` rather than `detect_` to emphasize the fact that this may modify `Config`.

Reply:
renamed